services/telemetry/config.go

```diff
@@ -0,0 +1,313 @@
+			timeout, err := time.ParseDuration(vmdb.TimeoutStr)
+			if err != nil {
+				return errors.Wrapf(err, "failed to parse duration [%s]", ds.VM.Timeout)
+			}
+			vmdb.Timeout = timeout
+		}
+	}
+
+	qandb := ds.QanDBSelect
+	if qandb.Enabled {
+		if qandb.TimeoutStr != "" {
+			timeout, err := time.ParseDuration(qandb.TimeoutStr)
+			if err != nil {
+				return errors.Wrapf(err, "failed to parse duration [%s]", ds.QanDBSelect.Timeout)
+			}
+			qandb.Timeout = timeout
+		}
+	}
+
+	pmmdb := ds.PmmDBSelect
+	if pmmdb.Enabled {
+		if pmmdb.TimeoutStr != "" {
+			timeout, err := time.ParseDuration(pmmdb.TimeoutStr)
+			if err != nil {
+				return errors.Wrapf(err, "failed to parse duration [%s]", ds.PmmDBSelect.Timeout)
+			}
+			pmmdb.Timeout = timeout
+		}
+	}
+
+	return nil
+}
+
+func (c *ServiceConfig) loadConfig(location string) ([]Config, error) { //nolint:cyclop
+	matches, err := filepath.Glob(location)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	var fileConfigs []FileConfig //nolint:prealloc
+	var fileCfg FileConfig
+	for _, match := range matches {
+		buf, err := ioutil.ReadFile(match) //nolint:gosec
+		if err != nil {
+			return nil, errors.Wrapf(err, "error while reading config [%s]", match)
+		}
+		if !c.DisableSigningVerification {
+			bufSign, err := ioutil.ReadFile(match + ".minisig") //nolint:gosec
+			if err != nil {
+				return nil, errors.Wrapf(err, "error while reading config [%s]", match)
+			}
+			valid := false
+			for _, publicKey := range c.Signing.trustedPublicKeysParsed {
+				if ok := minisign.Verify(publicKey, buf, bufSign); ok {
+					valid = true
+					break
+				}
+			}
+			if !valid {
+				return nil, errors.Errorf("signature verification failed for [%s]", match)
+			}
+		}
+		if err := yaml.Unmarshal(buf, &fileCfg); err != nil {
+			return nil, errors.Wrapf(err, "cannot unmashal config [%s]", match)
+		}
+		fileConfigs = append(fileConfigs, fileCfg)
+	}
+
+	if c.LoadDefaults {
+		defaultConfigBytes := []byte(defaultConfig)
+		if err := yaml.Unmarshal(defaultConfigBytes, &fileCfg); err != nil {
+			return nil, errors.Wrap(err, "cannot unmashal default config")
+		}
+		fileConfigs = append(fileConfigs, fileCfg)
+	}
+
+	if err := c.validateConfig(fileConfigs); err != nil {
+		c.l.Errorf(err.Error())
+	}
+
+	return c.merge(fileConfigs), nil
+}
+
+func (c *ServiceConfig) merge(cfgs []FileConfig) []Config {
+	var result []Config
+	ids := make(map[string]bool)
+	for _, cfg := range cfgs {
+		for _, each := range cfg.Telemetry {
+			_, exist := ids[each.ID]
+			if !exist {
+				ids[each.ID] = true
+				result = append(result, each)
+			}
+		}
+	}
+	return result
+}
+
+func (c *ServiceConfig) validateConfig(cfgs []FileConfig) error {
+	ids := make(map[string]bool)
+	for _, cfg := range cfgs {
+		for _, each := range cfg.Telemetry {
+			_, exist := ids[each.ID]
+			if exist {
+				return errors.Errorf("telemetry config ID duplication: %s", each.ID)
+			}
+			ids[each.ID] = true
+		}
+	}
+	return nil
```

Comment:
This fuction validates only duplicates, am rigth? Also it will return error on the first duplicate, and all other will not be logged. I guess we should also validate that ID, query, summary is not empty, source value is valid, etc...